src/playground/services/session/login/endpoint.clj

```diff
@@ -0,0 +1,33 @@
+(ns playground.services.session.login.endpoint
+  (:require [clojure.java.jdbc :as jdbc]
+            [honeysql.core :as h]
+            [clojure.spec.alpha :as spec]
+            [ring.util.response :as ring-resp]
+            [io.pedestal.http.route :refer [url-for]]
+            [buddy.hashers :as hashers]
+            [playground.services.session.login.logic :as logic]))
+
+(spec/def ::username (spec/and string? seq))
+(spec/def ::password (spec/and string? seq))
+(spec/def ::api (spec/keys :req-un [::username ::password]))
+
+(defn password-by-username [{:keys [db] :as request} username]
+  (let [db (->> db :pool (hash-map :datasource))]
+    (->> (logic/query-pass-by-user username)
+         (h/format)
+         (jdbc/query db))
+    ))
+
+(defn perform [request]
+  (let [username (get-in request [:form-params :username])
+        password (get-in request [:form-params :password])
+        session (:session request)]
+    (if (hashers/check password (:password (first (password-by-username request username))))
```

Comment:
The `hashers` stuff belongs to `logic` not `endpoint`. Generally, the endpoint code should be very thin: it only deals with HTTP stuff and invokes the "logic".

Think in these terms: if I created an alternative interface for this app (e.g., via a Desktop GUI instead of HTTP), would I have to duplicate any code?